Support constexpr std::count on non-Playstation builds

LumioseSil opened the pull request:
#### 70e99973935c2eb5716ba766ee239d279dfe4fa2
<pre>
Support constexpr std::count on non-Playstation builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=253157">https://bugs.webkit.org/show_bug.cgi?id=253157</a>

This does not compile on Playstation, so let us change the check
to conditionally check for those at compile time.

* Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp: Add preprocessor
  check against PlayStation before declaring size to be a const variable
  or constexpr.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba39c324bac36828f7da01cba41fbca1e8e63fa6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4899 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5024 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5196 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6415 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5013 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4892 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5198 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4996 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5256 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4984 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5064 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4386 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6432 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2552 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4382 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9344 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4414 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4456 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6051 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4875 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3967 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4375 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8440 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4736 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

foopoiuyt commented on Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp:
```diff
@@ -174,8 +174,13 @@ static constexpr SingleByteDecodeTable ibm866 {
     // Allocate this at runtime because building it at compile time would make the binary much larger and this is often not used.
-    // FIXME: With the C++20 version of std::count, we should be able to change this from const to constexpr and compute the size at compile time.
+    // FIXME: Remove this check after we use C++20 as a baseline
+#if __cplusplus >= 202002L
```
Unfortunately, I don't know this is going to be enough either because of cases where one has partial support for C++20 and where other parts of the code are relying on the parts that are supported (but this case is not).

In our case, we'll probably just end up having to add a `&& !PLATFORM(PLAYSTATION)` to the check, but I don't know if this might run afoul of anyone else.

mcatanzaro reviewed (changes requested):
Honestly I don't think it's worth changing this if it requires adding a preprocessor guard, so I suggest you abandon and close this pull request.

LumioseSil committed:
Support constexpr std::count on non-Playstation builds
https://bugs.webkit.org/show_bug.cgi?id=253157

This does not compile on Playstation, so let us change the check
to conditionally check for those at compile time.

* Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp: Add preprocessor
  check against PlayStation before declaring size to be a const variable
  or constexpr.

mcatanzaro reviewed (changes requested):
Again, this minor change is not worth the cost of adding a preprocessor guard.
